src/api/config.ts

```diff
@@ -49,3 +49,34 @@ config.post('/', async (req, res) => {
     logger.error(err);
   }
 });
+
+config.post('/bulk', async (req, res) => {
+  const { configKeys, configValues } = req.body;
+  // TODO: find a way to disable requireAuth() on the setup pages
+  // QUESTION: should I instead just zip the two arrays back into an object for simpler filtering?
+  // this solutions poses issues if two values are the same, but this is unlikely to occur in reality
+  const finalConfigItems: Config[] = [];
+
+  configValues
+    .filter((x: string) => x !== '')
+    .forEach(async (currentValue: string) => {
+      const currentKey = configKeys[configValues.indexOf(currentValue)];
+      try {
+        let configItem = await Config.findOne({ key: currentKey });
+        if (!configItem) {
+          configItem = new Config(currentKey, currentValue);
+        } else {
+          configItem.value = currentValue;
+        }
+
+        await configItem.save();
+        finalConfigItems.push(configItem);
+      } catch (err) {
+        /* istanbul ignore next */
+        res.status(500).send('Something went wrong sending an update to users; check the logs for more details');
+        /* istanbul ignore next */
+        logger.error(err);
+      }
+      res.send(finalConfigItems);
+    });
```

Comment:
We can send it as a single object so it's easier to loop over them. `Object.entries` takes an object and gives you an array of tuples containing the key and value. Also typeorm will automatically update the current entry in the database if the primary key is the same when you call `.save()` so you can do the following.

```suggestion
  const { config } = req.body;

  try {
    await Promise.all(Object.entries(config).map(([key, value]) => {
      const config = new Config(key, value);
      return await  config.save();
    }));

    res.sendStatus(200);
  } catch(err) {
    logger.error(err);
    res.status(500).send('Some items were not inserted, check logs for more information');
  }
```